src/Chapter2.hs

```diff
@@ -336,7 +336,17 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList a b arr
+  | a < 0 || b < 0 = []
```

Comment:
Your implementation is correct when the right bound is lower than the left. But I think you can implement it more efficiently by adding a separate case to your check :slightly_smiling_face:

Reply:
Are you talking about cases such as `a=-3` and `b=3` and you want to see the function output the values from `[0-3]` ? I had initially implemented it this way, but the tests were failing. So I assumed, this was an ignored case ? I'll maybe try to try to add that again :) Thank you !

Reply:
@neelchoudhury I mean cases when `a = 7` and `b = 3`. In this case, your implementation will take 4 elements from the list and drop 7 elements. This will be an empty list in the end, but this looks like unnecessary work if you can return an empty list straightway :fast_forward:

Reply:
I see. Thanks